src/todo-matcher.ts

```diff
@@ -0,0 +1,67 @@
+import { todoFilePathFor, todoFileNameFor } from './io';
+import { TodoData, TodoFilePathHash } from './types';
+
+export default class TodoMatcher {
+  private unprocessed: Set<TodoData>;
+
+  constructor() {
+    this.unprocessed = new Set();
+  }
+
+  unmatched(predicate: (todoDatum: TodoData) => boolean = () => false): string[] {
+    return [...this.unprocessed]
+      .filter((todoDatum) => predicate(todoDatum))
+      .map((todoDatum: TodoData) => {
+        return todoFilePathFor(todoDatum);
+      });
+  }
+
+  add(todoDatum: TodoData): void {
+    this.unprocessed.add(todoDatum);
+  }
+
+  find(todoFilePathHash: TodoFilePathHash): TodoData | undefined {
+    return [...this.unprocessed].find(
+      (todoDatum) => todoFileNameFor(todoDatum) === todoFilePathHash
+    );
+  }
+
+  exactMatch(todoDataToFind: TodoData): TodoData | undefined {
+    let found;
+
+    for (const todoDatum of this.unprocessed) {
+      if (
+        todoDataToFind.engine === todoDatum.engine &&
+        todoDataToFind.ruleId === todoDatum.ruleId &&
+        todoDataToFind.line === todoDatum.line &&
+        todoDataToFind.column === todoDatum.column
```

Comment:
Question: should this also be accounting for source once we have it?

Reply:
Ya let's add that. I'll add it in my next PR, which will add source support.